cromwell_tools/utilities.py

```diff
@@ -232,3 +238,95 @@ def validate_cromwell_label(label_object):
+                              collection_name=None, on_hold=False):
+    """Prepare the submission manifest for a workflow submission.
+
+    Args:
+        wdl_file (Union[str, io.BytesIO]): The workflow source file to submit for execution. Could be either the path
+            to the file (str) or the file content in io.BytesIO.
+        inputs_files (Optional[Union[List[Union[str, io.BytesIO]], str, io.BytesIO]]): The input data in JSON
+            format. Could be either the path to the file (str) or the file content in io.BytesIO. This could also
+            be a list of unlimited input file paths/contents, each of them should have a type of
+            Union[str, io.BytesIO]. (default None)
+        options_file (Optional[Union[str, io.BytesIO]]): The Cromwell options file for workflows. Could be either
+        the path to the file (str) or the file content in io.BytesIO. (default None)
+        dependencies (Optional[Union[str, List[str], io.BytesIO]]): Workflow dependency files. Could be the path to
+            the zipped file (str) containing dependencies, a list of paths(List[str]) to all dependency files to be
+            zipped or a zipped file in io.BytesIO. (default None)
+        label_file(Optional[Union[str, _io.BytesIO]]): JSON file containing a collection of key/value pairs for workflow
+            labels. (default None)
+        collection_name (Optional[str]): Collection in SAM that the workflow should belong to, if use CaaS. (
+            default None)
+        on_hold: (Optional[bool]) Whether to submit the workflow in "On Hold" status. (default False)
+
+    Returns
+        workflow_manifest (dict): A dictionary representing the workflow manifest ready for workflow submission.
+    """
+    workflow_manifest = {}
+
+    # Compose WDL source file
+    workflow_manifest['workflowSource'] = _download_to_BytesIO_if_string(wdl_file)
+
+    # Compose WDL inputs
+    if inputs_files:
+        if not isinstance(inputs_files, list):
+            inputs_files = [inputs_files]
+
+        for idx, inputs_file in enumerate(inputs_files):
+            if idx == 0:
+                # Compose WDL inputs 1
+                input_file_key = 'workflowInputs'
+            else:
+                # Compose other WDL inputs (from 2 - many)
+                input_file_key = 'workflowInputs_{X}'.format(X=idx+1)
+
+            workflow_manifest[input_file_key] = _download_to_BytesIO_if_string(inputs_file)
+
+    # Compose WDL options
+    if options_file:
+        workflow_manifest['workflowOptions'] = _download_to_BytesIO_if_string(options_file)
+
+    # Compose WDL labels
+    if label_file:
+        workflow_manifest['labels'] = _download_to_BytesIO_if_string(label_file)
+
+    # Compose WDL dependencies
+    if dependencies:
+        if isinstance(dependencies, list):
+            zip_file = make_zip_in_memory(download_to_map(dependencies))
+        else:
+            zip_file = _download_to_BytesIO_if_string(dependencies)
```

Comment:
How about adding a check to make sure this is a `.zip` file? If a user passes in `dependencies="/path/to/analysis.wdl"` and does not make it a list, it will get downloaded and not zipped.

Reply:
That is a great idea, which will make this function safer! Down the road, we could even use the `zipfile` tool to further validate the completeness of the zipfile. But for now, I will just add what you suggested!